### DISCUSSION TITLE
Flipello 10x10

### DISCUSSION THREAD

**Discussion by mtaktikos** [fairy-stockfish/Fairy-Stockfish#228] [Q&A] [answered]
Have added in the variants.ini :

[flipello10:flipello]
maxRank = 10
maxFile = 10
startFen = 10/10/10/10/4pP4/4Pp4/10/10/10/10[PPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPpppppppppppppppppppppppppppppppppppppppppppppppppppppp] w 0 1
enclosingDropStart = e5 f5 e6 f6

Have compiled a fresh Fairy-SF for large boards.
After the commands 
xboard  
variant flipello10
It shows the right FEN and draws the right board and pieces, but after
st 10
go
it starts with illegal moves like P@f3

Hope this is easy reproducible.
Could not find out, if it happens due to unflexible handling of the function "enclosingDropStart" or due to the unfixed problem with large boards.

**Comment by ianfab** [comment, marked as answer]
P@f3 is a legal move. This is due to the quirk of the CECP/XBoard protocol that rank numbering is 0-9 instead of 1-10 for games with exactly 10 ranks. Therefore, P@f3 in CECP is the same as P@f4 in UCI.

**Comment by mtaktikos** [comment]
I have a hot track on the problem above: I guess, the output of Fairy-SF P@f3 means P@f4, the output P@d5 means P@d6, and so on, i.e. Stockfish-intern the lowest left square is not a1, but a0

**Reply by mtaktikos** [reply]
Thx for the answer, Fabian
This quirk is not real a prob, now as I know that P@f3 is a legal move, it's possible to play 10x10 *thello with Fairy-SF

By the way, another help I need, to define also the last of alwey's variants:
Is it simple for you to implement a line to reverse the materialCounting after  stalematePieceCount, for example to multiplicate it with -1 ?
This would open an easy way for "anti reversi" variants, where the side with the lowest number of pieces wins

**Reply by ianfab** [reply]
Just to clarify, this is not because of Fairy-SF, it is the CECP that requires this behavior, and Fairy-SF just complies with it. So whether you need to use P@f3 or P@f4 depends on the protocol you choose (CECP or UCI).

Anti-Reversi should be feasible to implement. However, for feature requests and bug reports please use "issues", because "issues" are for topics that require code changes. This helps me to keep track of open tasks and relate them to commits in the repository when addressing them.
